### PR TITLE
add elasticsearch bims

### PR DESCRIPTION
**Pull request by ann26**
fix https://github.com/kartoza/LEDET_BIMS/issues/31

Library used:
- django-elasticsearch-dsl
- elasticsearch-dsl

### CODE REVIEW

File: bims/documents.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+biological_collection_record = Index('biological_collection_record')
+biological_collection_record.settings(
+    number_of_shards=1,
+    number_of_replicas=0
+)
+
+
+@biological_collection_record.doc_type
+class BiologicalCollectionRecordDocument(DocType):
+    original_species_name = fields.TextField()
```
Comment (lucernae) [inline bims/documents.py]: Try to omit this, because you already mentioned it at the bottom.

File: deployment/docker-compose.yml [hunk 2]
```diff
@@ -35,6 +35,16 @@ db:
 #  ports:
 #    - "25432:5432"
 
+elasticsearch:
+  # Note you cannot scale if you use conteiner_name
+  container_name: bims-elasticsearch
+  restart: on-failure:5
+  #  read_only: true
+  image: elasticsearch:2.4.2
```
Comment (lucernae) [inline deployment/docker-compose.yml]: I see at the bottom you used `elasticsearch-dsl>=5.0,<6.0`, so use `image: elasticsearch:5.6` for example (relevant images). If you actually want to use image elasticsearch:2.4.2, then perhaps your elasticsearch-dsl needs to be changed for version 2.x too.

File: deployment/docker/REQUIREMENTS.txt [hunk 3]
```diff
@@ -32,3 +32,5 @@ pygbif==0.2.0
 
 Pillow==5.1.0
 django-ordered-model==1.4.3
+django-elasticsearch-dsl
```
Comment (lucernae) [inline deployment/docker/REQUIREMENTS.txt]: pin the version if possible.

File: deployment/docker/REQUIREMENTS.txt [hunk 4]
```diff
@@ -32,3 +32,5 @@ pygbif==0.2.0
...
+elasticsearch-dsl>=5.0,<6.0
```
Comment (lucernae) [inline deployment/docker/REQUIREMENTS.txt]: Use the appropriate version according to your elastic search image.

**Comment by ann26** [comment]
Thank you @lucernae , it is working now! :)

**Comment by dimasciput** [comment]
@ann26 you need to put required library in setup.py, and please add instruction how to implement the elasticsearch

**Comment by ann26** [comment]
on front end:
peek 2018-06-07 17-32

File: bims/templates/map.html [hunk 5]
```diff
@@ -103,6 +106,27 @@
     <!-- Plugin JavaScript -->
     <script src="{% static "js/libs/jquery/jquery-3.3.1.min.js" %}"></script>
     <script src="{% static "js/libs/bootstrap-4.0.0/js/bootstrap.min.js" %}"></script>
+    <script>
```
Comment (dimasciput) [inline bims/templates/map.html]: @ann26 could you put this code on the js file?

**Comment by timlinux** [comment]
Can we have this merged before you go on leave please @ann26 ?

Thanks!

**Comment by ann26** [comment]
hi @timlinux ,
@dimasciput has been testing this PR and he said he wants to continue this work.

**Comment by timlinux** [comment]
> hi @timlinux ,
> @dimasciput has been testing this PR and he said he wants to continue this work.

ok thanks

**Comment by dimasciput** [comment]
@timlinux yes sorry, I think it still needs some improvements. The search results with fuzzy method is not very good, I've been tweaking with the configurations but still couldn't get the satisfactory result.

**Comment by lucernae** [comment]
Fuzzy is not very good. I've tried it in feti and I just prefer it to be turned off.

**Comment by dimasciput** [comment]
Continued in https://github.com/kartoza/django-bims/pull/31

**Pull request closed without merging**